### PR TITLE
Set tarfile.DEFAULT_FORMAT when generating bundle

### PR DESCRIPTION
**Pull request by murphm8**
This should fix https://github.com/colcon/colcon-bundle/issues/86

### CODE REVIEW

**Comment by murphm8** [comment]
We should considering moving entirely to PAX, benefits and support enumerated here: https://bugs.python.org/issue36268

**Comment by cottsay** [comment]
This still appears broken in `master`. Anything I can do to help move this along?

File: colcon_bundle/verb/bundlefile.py [hunk 1]
```diff
@@ -54,6 +54,13 @@ def add_overlay_archive(self, path):
 
     def close(self):  # noqa: N806
         """Close the archive."""
+        # We don't use any magic numbers here
+        # but Python 3.9 is updating to use the
+        # PAX format and we should do thorough
+        # testing before changing the format
+        # we use.
+        tarfile_format = tarfile.DEFAULT_FORMAT
```
Comment (thomas-moulard) [inline colcon_bundle/verb/bundlefile.py]: This code is not excption safe, I'd suggest writing it as follow:

https://docs.python.org/3/library/contextlib.html

```python
import contextlib

@contextlib.contextmanager
def _set_temporary_tarfile_default_format(value):
  tarfile_format = tarfile.DEFAULT_FORMAT
  tarfile.DEFAULT_FORMAT=  value
  try:
    yield
  finally:
    tarfile.DEFAULT_FORMAT = tarfile_format


  def close(...):
  # etc.
    with _set_temporary_tarfile_default_format(tarfile.GNU_FORMAT):
      # etc.
```

Comment (murphm8) [reply colcon_bundle/verb/bundlefile.py]: Updated to follow these guidelines.

**Pull request merged**